packages/beacon-node/src/api/impl/beacon/pool/index.ts

```diff
@@ -68,12 +68,21 @@ export function getBeaconPoolApi({
     },
 
     async getPoolAttesterSlashings() {
+      const fork = chain.config.getForkName(chain.clock.currentSlot);
+
+      if (isForkPostElectra(fork)) {
+        throw new ApiError(
+          400,
+          `Use getPoolAttesterSlashingsV2 to retrieve pool attester slashings for post-electra fork=${fork}`
+        );
+      }
+
       return {data: chain.opPool.getAllAttesterSlashings()};
     },
 
     async getPoolAttesterSlashingsV2() {
-      // TODO Electra: Determine fork based on data returned by api
-      return {data: chain.opPool.getAllAttesterSlashings(), meta: {version: ForkName.phase0}};
+      const fork = chain.config.getForkName(chain.clock.currentSlot);
```

Comment:
I think we can just use clock slot here and don't need to filter out phase0 attester slashings as those can be serialized using `electra.AttesterSlashing` ssz type and should still be able to include them onchain as well